### DISCUSSION TITLE
URL do WhatsApp NÃO abre

### DISCUSSION THREAD

**Discussion by NicolasHiga** [Flutterando/forum#170] [Perguntas e Respostas]
Fala pessoal, tô fazendo um projeto Flutter que há um botão onde o usuário poderia entrar em contato com os outros através do WhatsApp. Ao clicar, ele seria redirecionado.

Eu fiz esse código:
```
final Uri url = Uri.parse(https://wa.me/$userNumber*);

if
(!await launchUrl(ur1)) {
throw Exception(Could not launch $url');
}
```

Quando testei no emulador, não funciona e eu achava que era esse o problema. Porém, realizei alguns testes em dispositivos físicos e continua acontecendo a mesma coisa. Não abre de jeito nenhum, só da erro. Qualquer outro link funciona normalmente. E ele não chega a retornar um Exception, apenas abre a tela com a mensagem.

Não sei o que estou fazendo de errado. Esse link que estou utilizando é disponibilizado na própria documentação do WhatsApp. Até cheguei a tentar outras maneiras, vi vídeos, mas nada muda.

![image](https://github.com/Flutterando/forum/assets/108549497/25799d77-105a-4103-b443-97c3ef468c51)

**Comment by luizzlcs** [comment]
Tenta fazer assim :


 ```
void _launchUrl(String url) async {
      if (!await launch(
        url,
        forceWebView: false,
        enableJavaScript: true,
      )) {
        throw "Could not launch $url";
      }
    }
```


    No botão você chama o método:
    
 
```
ElevatedButon(
                  child: Text('WhastApp',
                  onTap: () => _launchUrl('https://wa.me/5584988369729'),
                ),
```
               
  Acho que isso deve funcionar.

**Reply by NicolasHiga** [reply]
Opa, deu certo aqui! Eu estava tentando abrir internamente. Precisava forçar abrir no navegador externo. Coloquei esse código:
```
if (!await launchUrl(url, mode: LaunchMode.externalApplication)) {
      throw Exception('Could not launch $url');
}
```
